tests/applepay_test.py

```diff
@@ -94,35 +149,414 @@ def test_signing_time_equals_current_time(token_fixture):
+    current_time = signing_time + timedelta(minutes=5)
     threshold = timedelta(hours=1)
 
     # When we attempt to validate the signing time against the threshold,
     with caplog.atLevel(logging.DEBUG):
-        applepay_utils.signing_time_is_valid(signature, current_time, threshold)
+        valid = applepay_utils.valid_signing_time(signing_time, current_time, threshold)
+
+    # Then the signign time is valid
+    assert valid is True
 
     # Then a new debug log is captured
-    records = caplog.records()
+    # filter on DEBUG log records only
+    records = filter(lambda log_record: log_record.levelno == logging.DEBUG, caplog.records())
     assert len(records) == 1
     assert records[0].name == 'applepay.utils'
-    assert records[0].message == 'Signing time is valid. Signing time: 2014-10-27 19:51:43 UTC+00:00, Current time: 2014-10-27 20:51:43 UTC, Threshold: 1:00:00.'
+    assert records[0].message == 'Signing time is valid. Signing time: 2014-10-27 20:51:43 UTC, Current time: 2014-10-27 20:56:43 UTC, Threshold: 1:00:00.'
 
 
-def test_invalid_signing_time_data_is_logged(token_fixture, caplog):
-    # Given: a invalid signature for a current time and threshold
-    signature = token_fixture['signature']
-    current_time = datetime(2010, 1, 2, 5, 22, 13, tzinfo=utc)
+def test_invalid_signing_time_data_is_logged(caplog):
+    # Given: an invalid signing time
+    signing_time = datetime(2014, 10, 27, 20, 51, 43, tzinfo=utc)
+    current_time = signing_time + timedelta(hours=5)
     threshold = timedelta(hours=1)
 
     # When we attempt to validate the signing time against the threshold,
     with caplog.atLevel(logging.DEBUG):
-        applepay_utils.signing_time_is_valid(signature, current_time, threshold)
+        valid = applepay_utils.valid_signing_time(signing_time, current_time, threshold)
+
+    # Then the signing time is not valid
+    assert valid is False
 
     # Then a new debug log is captured
-    records = caplog.records()
+    # filter on DEBUG log records only
+    records = filter(lambda log_record: log_record.levelno == logging.DEBUG, caplog.records())
     assert len(records) == 1
     assert records[0].name == 'applepay.utils'
-    assert records[0].message == 'Signing time is invalid. Signing time: 2014-10-27 19:51:43 UTC+00:00, Current time: 2010-01-02 05:22:13 UTC, Threshold: 1:00:00.'
+    assert records[0].message == 'Signing time is invalid. Signing time: 2014-10-27 20:51:43 UTC, Current time: 2014-10-28 01:51:43 UTC, Threshold: 1:00:00.'
+
+
+def test_verify_signature(token_fixture):
+    """Test that a token known to be valid has a valid
+    signature"""
+    assert applepay_utils.verify_signature(token_fixture)
+
+
+def test_verify_signature_with_threshold(token_fixture):
+    """Test that a token known to be valid has a valid
+    signature and the signing time is within the provided threshold"""
+    assert applepay_utils.verify_signature(token_fixture, threshold=timedelta(days=1000))
```

Comment:
Yea not sure what I was thinking here.  I am going to add https://pypi.python.org/pypi/freezegun as a testing requirement in order to control datetime.now.